internal/elasticsearch/security/api_key.go

```diff
@@ -155,6 +187,36 @@ func resourceSecurityApiKeyCreate(ctx context.Context, d *schema.ResourceData, m
 	return resourceSecurityApiKeyRead(ctx, d, meta)
 }
 
+func doesCurrentVersionSupportRestrictionOnApiKey(ctx context.Context, client *clients.ApiClient, keysWithRestrictions []string) (isSupported bool, err error) {
+	if esClient, err := client.GetESClient(); err == nil {
+		req := esapi.InfoRequest{}
+		res, err := req.Do(ctx, esClient.Transport)
+		if err != nil {
+			return false, err
+		}
+
+		defer res.Body.Close()
+
+		var info info
+		err = json.NewDecoder(res.Body).Decode(&info)
+		if err != nil {
+			return false, err
+		}
+
+		currentVersion, err := version.NewVersion(info.Version.Number)
+		if err != nil {
+			return false, err
+		}
+
+		supportedVersion, _ := version.NewVersion("8.9.0") // restriction is supported since 8.9.x
+
+		if currentVersion.LessThan(supportedVersion) {
+			return false, err
+		}
+	}
+	return true, nil
```

Comment:
Can replace this with something like

```suggestion
	currentVersion, diags := client.ServerVersion()
	if diags.HasError() {
		return true
	}
	
	return currentVersion.GreaterThanOrEqual(APIKeyWithRestrictionMinVersion)
```

Reply:
Let me give that a go